pkg/graveler/graveler.go

```diff
@@ -1851,8 +1846,51 @@ func (g *KVGraveler) stagingEmpty(ctx context.Context, repositoryID RepositoryID
 	return true, nil
 }
 
+// dropStaging deletes all staging area entries of a given branch from store
+// We do not wait for result as this is an asynchronous operation
+func (g *KVGraveler) dropStaging(ctx context.Context, branch *Branch) {
+	tokens := branch.SealedTokens
+	tokens = append(tokens, branch.StagingToken)
+	for _, st := range tokens {
+		token := st // pinning
+		go func() {
+			err := g.StagingManager.Drop(ctx, token)
```

Comment:
I think the error should be returned, why shadow it from the caller?

Reply:
Once we update the branch for all things considered we've reset the branch. Waiting for the error to be returned will prolong the operation and return an error which is not interesting for the user